pkg/config/v3/config.go

```diff
@@ -0,0 +1,349 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v3
+
+import (
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+)
+
+// Version is the config.Version for project configuration 3
+var Version = config.Version{Number: 3}
+
+type cfg struct {
+	// Version
+	Version config.Version `json:"version"`
+
+	// String fields
+	Domain     string `json:"domain,omitempty"`
+	Repository string `json:"repo,omitempty"`
+	Name       string `json:"projectName,omitempty"`
+	Layout     string `json:"layout,omitempty"`
+
+	// Boolean fields
+	MultiGroup      bool `json:"multigroup,omitempty"`
+	ComponentConfig bool `json:"componentConfig,omitempty"`
+
+	// Resources
+	Resources []resource.Resource `json:"resources,omitempty"`
+
+	// Plugins
+	Plugins PluginConfigs `json:"plugins,omitempty"`
+}
+
+// PluginConfigs holds a set of arbitrary plugin configuration objects mapped by plugin key.
+// TODO: do not export this once internalconfig has merged with config
+type PluginConfigs map[string]pluginConfig
+
+// pluginConfig is an arbitrary plugin configuration object.
+type pluginConfig interface{}
+
+// New returns a new config.Config
+func New() config.Config {
+	return &cfg{Version: Version}
+}
+
+func init() {
+	config.Register(Version, New)
+}
+
+// GetVersion implements config.Config
+func (c cfg) GetVersion() config.Version {
+	return c.Version
+}
+
+// GetDomain implements config.Config
+func (c cfg) GetDomain() string {
+	return c.Domain
+}
+
+// SetDomain implements config.Config
+func (c *cfg) SetDomain(domain string) error {
+	c.Domain = domain
+	return nil
+}
+
+// GetRepository implements config.Config
+func (c cfg) GetRepository() string {
+	return c.Repository
+}
+
+// SetRepository implements config.Config
+func (c *cfg) SetRepository(repository string) error {
+	c.Repository = repository
+	return nil
+}
+
+// GetProjectName implements config.Config
+func (c cfg) GetProjectName() string {
+	return c.Name
+}
+
+// SetProjectName implements config.Config
+func (c *cfg) SetProjectName(name string) error {
+	c.Name = name
+	return nil
+}
+
+// GetLayout implements config.Config
+func (c cfg) GetLayout() string {
+	return c.Layout
+}
+
+// SetLayout implements config.Config
+func (c *cfg) SetLayout(layout string) error {
+	c.Layout = layout
+	return nil
+}
+
+// IsMultiGroup implements config.Config
+func (c cfg) IsMultiGroup() bool {
+	return c.MultiGroup
+}
+
+// SetMultiGroup implements config.Config
+func (c *cfg) SetMultiGroup() error {
+	c.MultiGroup = true
+	return nil
+}
+
+// ClearMultiGroup implements config.Config
+func (c *cfg) ClearMultiGroup() error {
```

Comment:
cannot the set receive a boolean instead of we need 2 methods for that?

Reply:
It could, it just was defined the other way round when I created the `Config` API